parser/abstract_scraper.py

```diff
@@ -0,0 +1,64 @@
+import logging
+from abc import ABC, abstractmethod
+from parser.parser_exceptions import ApartmentLinkNotFoundException, ApartmentTitleNotFoundException
+from parser.apartment import Apartment
+from typing import Any, Optional
+
+
+class AbstractScraper(ABC):
+    ApartmentPageData = Any
```

Comment:
Usually types are not a part of class namespaces. And i don't feel it should be here like this. So please put it out of the class